src/main/java/com/google/sps/data/FirestoreDB.java

```diff
@@ -64,58 +64,60 @@ public void deleteUser(String userId) {
-          userRef.update(Collections.singletonMap(nestedPropertyName, true));
-      DBUtils.blockOnFuture(addToCreatedPostsFuture);
-    }
-    
-    /** Remove a recipe from a user's created (when recipe is deleted). */
-    public void removeRecipeIdFromCreated(String userId, String recipeId) {
-      DocumentReference userRef = DBUtils.user(userId);
-      String nestedPropertyName = DBUtils.getNestedPropertyName(User.CREATED_RECIPES_KEY, recipeId);
-      Map<String, Object> update = new HashMap<>();
-      update.put(nestedPropertyName, FieldValue.delete());
-      DBUtils.blockOnFuture(userRef.update(update));
-    }
-    
-    /** Add recipe id: true as a singleton map to user's saved recipes field. */
-    public void addRecipeIdToSaved(String userId, String recipeId) {
-      DocumentReference userRef = DBUtils.user(userId);
-      String nestedPropertyName = DBUtils.getNestedPropertyName(User.SAVED_RECIPES_KEY, recipeId);
-      ApiFuture addToSavedPostsFuture =
+      String nestedPropertyName;
+      
+      switch (collection) {
+        case User.SAVE:
+          nestedPropertyName = DBUtils.getNestedPropertyName(User.SAVED_RECIPES_KEY, objectId);
+          break;
+        case User.CREATE:
+          nestedPropertyName = DBUtils.getNestedPropertyName(User.CREATED_RECIPES_KEY, objectId);
+          break;
+        case User.TAG:
+          nestedPropertyName = DBUtils.getNestedPropertyName(User.TAGS_FOLLOWED_KEY, objectId);
+          break;
+      }
+
+      ApiFuture addUserPropertyFuture = 
           userRef.update(Collections.singletonMap(nestedPropertyName, true));
-      DBUtils.blockOnFuture(addToSavedPostsFuture);
+      DBUtils.blockOnFuture(addUserPropertyFuture);
     }
 
-    /** Remove a recipe from a user's created (when recipe is deleted or unsaved). */
-    public void removeRecipeIdFromSaved(String userId, String recipeId) {
+    /**
+     * Deletes a property value for a certain user document.
+     * Can be used to let a user delete a recipe from saved or created, or to let user unfollow a tag.
+     * @param userId the user's Firebase ID
+     * @param objectId the ID of either a recipe if the intent is to unsave/create, or of a tag for tag unfollowing.
+     * @param collection a constant from User class indicating which mode -- save, create, or tag.
+     */
+    public void deleteUserProperty(String userId, String objectId, String collection) {
       DocumentReference userRef = DBUtils.user(userId);
-      String nestedPropertyName = DBUtils.getNestedPropertyName(User.SAVED_RECIPES_KEY, recipeId);
-      Map<String, Object> update = new HashMap<>();
-      update.put(nestedPropertyName, FieldValue.delete());
-      DBUtils.blockOnFuture(userRef.update(update));
-    }
+      String nestedPropertyName;
+      
+      switch (collection) {
+        case User.SAVE:
+          nestedPropertyName = DBUtils.getNestedPropertyName(User.SAVED_RECIPES_KEY, objectId);
+          break;
```

Comment:
Use `nestedPropertyName = DBUtils.getNestedPropertyName(collection, objectId);` instead of this switch statement seeing as there are already constants for those collections.